library/proc_macro/src/diagnostic.rs

```diff
@@ -1,182 +1,238 @@
 use crate::Span;
+use std::iter;
 
 /// An enum representing a diagnostic level.
-#[unstable(feature = "proc_macro_diagnostic", issue = "54140")]
+#[doc(hidden)]
+#[unstable(feature = "proc_macro_internals", issue = "27812")]
 #[derive(Copy, Clone, Debug)]
 #[non_exhaustive]
 pub enum Level {
     /// An error.
     Error,
-    /// A warning.
-    Warning,
+    // /// A warning.
+    // Warning,
```

Comment:
I'm not a fan of keeping commented code around. On first read I thought this was a typo (or bad merge).

Can we replace this with only a TODO that links to the relevant issue?

Same with the `fn warning` definition below.

Reply:
Yeah, probably best to just remove it outright, given that the form lint-associated warnings take will certainly require a different API.